src/core/server/saved_objects/permission_control/client.ts

```diff
@@ -86,27 +93,53 @@ export class SavedObjectsPermissionControl {
     permissionModes: SavedObjectsPermissionModes
   ) {
     const savedObjectsGet = await this.bulkGetSavedObjects(request, savedObjects);
-    if (savedObjectsGet) {
-      const principals = this.getPrincipalsFromRequest(request);
-      const hasAllPermission = savedObjectsGet.every((item) => {
-        // for object that doesn't contain ACL like config, return true
-        if (!item.permissions) {
-          return true;
-        }
-        const aclInstance = new ACL(item.permissions);
-        return aclInstance.hasPermission(permissionModes, principals);
-      });
+    if (!savedObjectsGet) {
       return {
-        success: true,
-        result: hasAllPermission,
+        success: false,
+        error: i18n.translate('savedObjects.permission.notFound', {
+          defaultMessage: 'Can not find target saved objects.',
+        }),
       };
     }
 
+    if (savedObjectsGet.length === 1 && !!savedObjectsGet[0].error) {
+      return {
+        success: false,
+        error: savedObjectsGet[0].error,
+      };
+    }
+
+    const principals = this.getPrincipalsFromRequest(request);
+    const hasAllPermission = savedObjectsGet.every((item) => {
+      // for object that doesn't contain ACL like config, return true
+      if (!item.permissions) {
+        return true;
+      }
+      const aclInstance = new ACL(item.permissions);
+      return aclInstance.hasPermission(permissionModes, principals);
+    });
+    if (!hasAllPermission) {
+      const savedObjectsBasicInfo = savedObjectsGet.map((item) => {
+        return {
+          id: item.id,
+          type: item.type,
+          workspaces: item.workspaces,
+          permissions: item.permissions,
+        };
+      });
+      this.logger.debug(
```

Comment:
We can use [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) which is easier to use

Reply:
I've changed that, thanks!